src/screencast/pipewire_screencast.c

```diff
@@ -170,28 +172,59 @@ static void pwr_handle_stream_add_buffer(void *data, struct pw_buffer *buffer) {
 	d = buffer->buffer->datas;
 
 	// Select buffer type from negotiation result
-	if ((d[0].type & (1u << SPA_DATA_MemPtr)) > 0) {
-		d[0].type = SPA_DATA_MemPtr;
+	if ((d[0].type & (1u << SPA_DATA_MemFd)) > 0) {
+		d[0].type = SPA_DATA_MemFd;
 	} else {
 		logprint(ERROR, "pipewire: unsupported buffer type");
 		cast->err = 1;
 		return;
 	}
 
+	logprint(TRACE, "pipewire: selected buffertype %d", d[0].type);
 	// Prepare buffer for choosen type
-	if (d[0].type == SPA_DATA_MemPtr) {
+	if (d[0].type == SPA_DATA_MemFd) {
+		d[0].type = SPA_DATA_MemFd;
 		d[0].maxsize = cast->simple_frame.size;
 		d[0].mapoffset = 0;
 		d[0].chunk->size = cast->simple_frame.size;
 		d[0].chunk->stride = cast->simple_frame.stride;
 		d[0].chunk->offset = 0;
-		d[0].flags = 0;
-		d[0].fd = -1;
+		d[0].flags = SPA_DATA_FLAG_READWRITE;
```

Comment:
Shouldn't we make the buffer read-only for our consumers?

Reply:
Do we care about another client writing in our buffer stuff? I think it should be overwritten by the next screencopy. 
This might help with filter, which want to avoid the overhead of copying every buffer to another one. But we can just set no flag (I think i took that just from the pipewire example).

Reply:
I'd prefer to avoid setting the buffer as read-write, to allow future optimizations like handling out the same frames to multiple clients.

Reply:
done